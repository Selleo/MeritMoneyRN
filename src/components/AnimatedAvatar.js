import React, { Component } from 'react'
import { View, StyleSheet } from 'react-native'
import * as Animatable from 'react-native-animatable'

import consumer from 'src/hocs/consumer'
import { primaryLight } from 'src/styles/colors'
import Avatar from './Avatar'

export class AnimatedAvatar extends Component {
  state = {
    visible: true,
  }

  shouldComponentUpdate = nextProps =>
    this.props.avatarAnimationValue !== nextProps.avatarAnimationValue

  componentDidUpdate = () => {
    this.animate(this.props.avatarAnimationValue)
  }

  animate = visible => this.setState({ visible: !visible })

  render() {
    return (
      <Animatable.View
        animation={this.state.visible ? 'zoomIn' : 'zoomOut'}
        duration={300}
        style={styles.avatarContainer}
      >
        <View style={styles.avatarBorder}>
          <Avatar size={80} />
        </View>
      </Animatable.View>
    )
  }
}

export default consumer(AnimatedAvatar)

const styles = StyleSheet.create({
  avatarContainer: {
    alignSelf: 'center',
    position: 'absolute',
    top: 25,
    zIndex: 2,
  },
  avatarBorder: {
    borderRadius: 45,
    height: 90,
    width: 90,
    justifyContent: 'center',
    alignItems: 'center',
    borderWidth: 5,
    borderColor: primaryLight,
    zIndex: 3,
  },
})
