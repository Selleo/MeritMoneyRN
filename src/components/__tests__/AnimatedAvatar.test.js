import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import { AnimatedAvatar } from '../AnimatedAvatar'

describe('<AnimatedAvatar />', () => {
  const defaultProps = {
    avatarAnimationValue: 1,
  }

  const wrapper = renderer.create(<AnimatedAvatar {...defaultProps} />)

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('componentDidUpdate', () => {
    test('should call animate if animation value has changed', () => {
      const wrapper = renderer.create(<AnimatedAvatar {...defaultProps} />)
      const instance = wrapper.getInstance()
      instance.animate = jest.fn()
      wrapper.update(<AnimatedAvatar avatarAnimationValue={0} />)
      expect(instance.animate).toHaveBeenCalled()
    })

    test('should not call animate if animation value has not changed', () => {
      const wrapper = renderer.create(<AnimatedAvatar {...defaultProps} />)
      const instance = wrapper.getInstance()
      instance.animate = jest.fn()
      wrapper.update(<AnimatedAvatar avatarAnimationValue={1} />)
      expect(instance.animate).not.toHaveBeenCalled()
    })
  })

  describe('animate', () => {
    test('should change animation value', () => {
      const instance = wrapper.getInstance()
      instance.state.visible = true
      instance.animate(true)
      expect(instance.state.visible).toBe(false)
    })
  })
})
