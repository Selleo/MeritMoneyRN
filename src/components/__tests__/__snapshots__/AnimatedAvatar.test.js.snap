// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimatedAvatar /> render 1`] = `"View"`;
